Source/cielim/Private/SimulationDataSourceActor.cpp

```diff
@@ -183,7 +183,13 @@ void ASimulationDataSourceActor::NetworkTick(float DeltaTime)
 		this->ShouldUpdateScene = true;
 		UE_LOG(LogCielim, Display, TEXT("Reading sim update data: ASimulationDataSourceActor"));
 	} else if (std::holds_alternative<RequestImage>(QueueData.value().Query)) {
-		this->NetworkSimulationDataSource->PutImageQueueData(this->CaptureManager->GetPNG());
+		double pointSpread = (double) this->CielimMessage.camera().pointspreadfunction();
+		double readNoise = this->CielimMessage.camera().readnoise();
+		double systemGain = this->CielimMessage.camera().systemgain();
+		// Using cosmic ray std deviation as number of rays at the moment
+		int cosmicRayStdDev = (int) this->CielimMessage.camera().renderparameters().cosmicraystddeviation();
+		
+		this->NetworkSimulationDataSource->PutImageQueueData(this->CaptureManager->GetPNG(pointSpread, readNoise, systemGain, cosmicRayStdDev));
```

Comment:
I think this could be the best place to check if not None, and have default values otherwise.
Essentially, no blur, no noise, systemGain of 1 and no cosmic Rays would be the defaults

Reply:
This has been done in the render functions to avoid cluttering the getPNG function.